app/views/dev_mode.py

```diff
@@ -91,19 +92,14 @@ def dev_flush_mode_get():
 
 def extract_eq_id_and_form_type(schema_name):
     logger.debug("extracting eq_id and form type from schema name", schema_name=schema_name)
-    if "_" in schema_name:
-        split_schema_name = schema_name.split("_", 1)
-        if len(split_schema_name) != 2:
-            raise ValueError("Schema file name incorrect %", schema_name)
+    if re.search(r"^([a-z0-9]+)_(\w+)\.json$", schema_name):
+        split_schema_name = schema_name.split("_", maxsplit=1)
```

Comment:
Looking good, would be better to use re.match here then you can extract the values that matched directly e.g.

```python
match = re.match(r"^([a-z0-9]+)_(\w+)\.json$", schema_name)
if match:
    eq_id = match.group(1)
    form_type = match.group(2)
```